cmd/kube-apiserver/app/server.go

```diff
@@ -251,6 +222,43 @@ func CreateKubeAPIServerConfig(s *options.ServerRunOptions) (*master.Config, inf
 		Dial:            proxyDialerFn,
 		TLSClientConfig: proxyTLSClientConfig,
 	})
+	return nodeTunneler, proxyTransport, nil
```

Comment:
I assume you are referring to the line "proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}". While the transport which contains this config is passed to the Aggregator, that flag is never actually picked up by the Aggregator. With this change the Aggregator only actually uses the proxyDialerFn field from the Transport (and not the config).  @lavalamp and I were planning to do a further code clean up to pass something with more specific funcationality, so as adding a SSHTunnel to an existing Transport, and isolating that logic appropriately.  That should both make the code cleaner and make the code more resistant to something like accidentally inheriting the  InsecureSkipVerify field. Currently that field is set in the Aggregator, so if you do hit the bug referenced in the comment ("Proxying to pods and services is IP-based... don't expect to be able to verify the hostname") on the Aggregator; It would be worked around by setting that flag on the Aggregator.